core/shared/src/test/scala/fs2/ChunkSpec.scala

```diff
@@ -19,101 +24,110 @@ class ChunkSpec extends Fs2Spec {
-    "size.double" in propSize[Double, Chunk[Double]]
-    "size.long" in propSize[Long, Chunk[Long]]
-    "size.unspecialized" in propSize[Int, Chunk[Int]]
-
-    "take.boolean" in propTake[Boolean, Chunk[Boolean]]
-    "take.byte" in propTake[Byte, Chunk[Byte]]
-    "take.byteBuffer" in propTake[Byte, Chunk.ByteBuffer]    
-    "take.double" in propTake[Double, Chunk[Double]]
-    "take.long" in propTake[Long, Chunk[Long]]
-    "take.unspecialized" in propTake[Int, Chunk[Int]]
-
-    "drop.boolean" in propDrop[Boolean, Chunk[Boolean]]
-    "drop.byte" in propDrop[Byte, Chunk[Byte]]
-    "drop.byteBuffer" in propDrop[Byte, Chunk.ByteBuffer]
-    "drop.double" in propDrop[Double, Chunk[Double]]
-    "drop.long" in propDrop[Long, Chunk[Long]]
-    "drop.unspecialized" in propDrop[Int, Chunk[Int]]
-
-    "isempty.boolean" in propIsEmpty[Boolean, Chunk[Boolean]]
-    "isempty.byte" in propIsEmpty[Byte, Chunk[Byte]]
-    "isempty.byteBuffer" in propIsEmpty[Byte, Chunk.ByteBuffer]
-    "isempty.double" in propIsEmpty[Double, Chunk[Double]]
-    "isempty.long" in propIsEmpty[Long, Chunk[Long]]
-    "isempty.unspecialized" in propIsEmpty[Int, Chunk[Int]]
-
-    "toarray.boolean" in propToArray[Boolean, Chunk[Boolean]]
-    "toarray.byte" in propToArray[Byte, Chunk[Byte]]
-    "toarray.byteBuffer" in propToArray[Byte, Chunk.ByteBuffer]
-    "toarray.double" in propToArray[Double, Chunk[Double]]
-    "toarray.long" in propToArray[Long, Chunk[Long]]
-    "toarray.unspecialized" in propToArray[Int, Chunk[Int]]
-
-    "tobytebuffer.byte" in propToByteBuffer[Chunk[Byte]]
-    "tobytebuffer.byteBuffer" in propToByteBuffer[Chunk.ByteBuffer]
-
-    // "map.boolean => byte" in forAll { c: Chunk[Boolean] =>
-    //   (c map { b => if (b) 0.toByte else 1.toByte }).toArray shouldBe (c.toArray map { b => if (b) 0.toByte else 1.toByte })
-    // }
-
-    // "map.long => long" in forAll { c: Chunk[Long] => (c map (1 +)).toArray shouldBe (c.toArray map (1 +)) }
+  implicit val arbDoubleChunk: Arbitrary[Chunk[Double]] = Arbitrary {
+    for {
+      n <- Gen.choose(0, 100)
+      values <- Gen.containerOfN[Array, Double](n, Arbitrary.arbDouble.arbitrary)
+      offset <- Gen.choose(0, n)
+      sz <- Gen.choose(0, n - offset)
+    } yield Chunk.doubles(values, offset, sz)
   }
+
+  def testChunk[A: Arbitrary: ClassTag: Cogen: CommutativeMonoid: Eq](name: String, of: String, testTraverse: Boolean = true)(implicit C: Arbitrary[Chunk[A]]): Unit = {
+    s"$name" - {
+      "size" in propSize[A, Chunk[A]]
+      "take" in propTake[A, Chunk[A]]      
+      "drop" in propDrop[A, Chunk[A]]
+      "isempty" in propIsEmpty[A, Chunk[A]]
+      "toarray" in propToArray[A, Chunk[A]]
+
+      if (implicitly[ClassTag[A]] == ClassTag.Byte)
+        "tobytebuffer.byte" in propToByteBuffer[Chunk[Byte]]
+
+      checkAll(s"Eq[Chunk[$of]]", EqTests[Chunk[A]].eqv)
+      checkAll(s"Monad[Chunk]", MonadTests[Chunk].monad[A, A, A])
+
+      if (testTraverse)
```

Comment:
I'm not proud of this or the conditional above.  Maybe `testChunk` is too fancy.  If we were to repeat these tests for each specialization, we might want to formalize `ChunkLaws` in the discipline style.